backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -293,6 +302,10 @@ public void dataHubUploaderTask() {
             return;
         }
 
+        if (!_dataHubUploadRepo.tryUploadLock()) { // take the advisory lock for this process
```

Comment:
I put this after the sanity checks and you put it before them—I think yours is probably safer but it's a bit silly to do it twice 😁